packages/blade/src/components/Input/OTPInput/OTPInput.tsx

```diff
@@ -0,0 +1,270 @@
+import React, { useEffect, useState } from 'react';
+import type { BaseInputProps } from '../BaseInput';
+import { BaseInput } from '../BaseInput';
+import { getHintType } from '../BaseInput/BaseInput';
+import type { FormInputOnEvent } from '../../Form';
+import { FormHint, FormLabel } from '../../Form';
+import { useFormId } from '../../Form/useFormId';
+import type { FormInputOnKeyDownEvent } from '../../Form/FormTypes';
+import Box from '~components/Box';
+import { getPlatformType } from '~utils';
+
+export type OTPInputProps = Pick<
+  BaseInputProps,
+  | 'label'
+  | 'labelPosition'
+  | 'validationState'
+  | 'helpText'
+  | 'errorText'
+  | 'successText'
+  | 'name'
+  | 'onChange'
+  | 'value'
+  | 'isDisabled'
+  | 'autoFocus'
+  | 'keyboardReturnKeyType'
+  | 'keyboardType'
+  | 'placeholder'
+> & {
+  /**
+   * Determines the number of input fields to show for the OTP
+   * @default 6
+   */
+  otpLength?: 4 | 6;
+  /**
+   * The callback function to be invoked when all the values of the OTPInput are filled
+   */
+  onOTPFilled?: FormInputOnEvent;
+};
+
+const isReactNative = getPlatformType() === 'react-native';
+
+/**
+ *  Converts a string value of otp to array if passed otherwise returns an array of 6 empty strings
+ */
+const otpToArray = (code?: string): string[] => code?.split('') ?? Array(6).fill('');
+
+const OTPInput = ({
```

Comment:
Maybe we can add some JSDoc with basic minimum usage example?

Reply:
Good idea! Added!